### PR TITLE
sessions: add smoke test

### PR DESCRIPTION
**Pull request by hppritcha**
and add to mac os github CI

Signed-off-by: Howard Pritchard <howardp@lanl.gov>
(cherry picked from commit 9109c33bb4bfa6d7788e2add651f192a678b1334)

we need this PR before another one with a bug fix can be pushed to v5.0.x

### CODE REVIEW

File: examples/hello_sessions_c.c [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * functionality.  See sections 11.3 and 11.4 of the MPI 4 standard for more
+ * details.
+ */
+
+int main(int argc, char** argv) {
+  MPI_Info info;
+  MPI_Session s1, s2;
+
+#if 0
+/* need PR https://github.com/open-mpi/ompi/pull/12868 to be merged in
```
Comment (jsquyres) [inline examples/hello_sessions_c.c:15]: It looks like https://github.com/open-mpi/ompi/pull/12868 has already been merged.

File: examples/hello_sessions_c.c [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * before this can be uncommented.
+ */
+  MPI_Info_create(&info);
+#endif
+  MPI_Session_init(MPI_INFO_NULL, MPI_ERRORS_RETURN, &s1);
+  MPI_Session_finalize(&s1);
+  MPI_Session_init(MPI_INFO_NULL, MPI_ERRORS_RETURN, &s2);
+  MPI_Session_finalize(&s2);
+}
```
Comment (jsquyres) [inline examples/hello_sessions_c.c:24]: Should put a `return 0` here.

Comment (hppritcha) [reply examples/hello_sessions_c.c:24]: The cherry-pick of https://github.com/open-mpi/ompi/pull/12888 back to v5.0.x will address that first comment.
I'll add the return 0 in a PR to main first.  Maybe I'll cherry-pick the commit from #12888 on to this PR.

**Comment by hppritcha** [comment]
this PR will not pass CI until https://github.com/open-mpi/ompi/pull/12883 is merged into the v5.0.x branch.

File: examples/hello_sessions_c.c (hunk 2 above)
Comment (hppritcha) [reply examples/hello_sessions_c.c:21]: return 0 added

**Review by jsquyres — APPROVED** [review]

**Comment by jsquyres** [comment]
Amusing -- your new test is failing on this PR.  😆

**Comment by janjust** [comment]
> this PR will not pass CI until #12883 is merged into the v5.0.x branch.

done, merged

**Review by janjust — APPROVED** [review]

**Pull request merged**